frontend/src/components/ScheduledEventsTableRow/ScheduledEventsTableRow.tsx

```diff
@@ -0,0 +1,59 @@
+import * as React from 'react';
+
+export interface ScheduledEventsTableRowProps {
+  id: number;
+  name: string;
+  description: string;
+  enabled: boolean;
+  interval: number;
+  repeat: string;
+  interval_type: string;
+  start_date: string;
+}
+
+export const ScheduledEventsTableRow: React.SFC<ScheduledEventsTableRowProps> = (props) => {
+  const derivedTime = () => {
+    const dt = new Date();
+    dt.setTime(Date.parse(props.start_date));
+    let hours = dt.getUTCHours();
+    const AmOrPm = hours >= 12 ? 'PM' : 'AM';
+    hours = hours % 12 || 12;
+    const minutes = dt.getUTCMinutes();
+    const finalTime = hours + ':' + (minutes < 10 ? '0' + minutes : minutes) + ' ' + AmOrPm;
+
+    return finalTime;
+  };
+
+  const derivedDate = () => {
+    const day = new Date();
+    day.setTime(Date.parse(props.start_date));
+    const dd = String(day.getDate()).padStart(2, '0');
+    const mm = String(day.getMonth() + 1).padStart(2, '0');
+    const yyyy = day.getFullYear();
+    const newDate = dd + '-' + mm + '-' + yyyy;
+
+    return newDate;
+  };
```

Comment:
Move these out of the component; give them proper names based on what they do, and a proper return type.